pam/pam_test/utils.go

```diff
@@ -0,0 +1,60 @@
+// Package pam_test includes Test tools for the PAM module
+package pam_test
+
+/*
+#include <stdlib.h>
+
+#ifdef __SANITIZE_ADDRESS__
+#include <sanitizer/lsan_interface.h>
+#endif
+
+static inline void
+maybe_do_leak_check (void)
+{
+#ifdef __SANITIZE_ADDRESS__
+	__lsan_do_leak_check();
+#endif
+}
+*/
+import "C"
+
+import (
+	"os"
+	"runtime"
+	"time"
+	"unsafe"
+
+	"github.com/msteinert/pam"
+)
+
+// MaybeDoLeakCheck triggers the garbage collector and if the go program is
+// compiled with -asan flag, do a memory leak check.
+// This is meant to be used as a test Cleanup function, to force Go detecting
+// if allocated resources have been released, e.g. using
+// t.Cleanup(pam_test.MaybeDoLeakCheck)
+func MaybeDoLeakCheck() {
```

Comment:
This is part of this package and only used there, so no need to export it?

Reply:
Well, it's used only inside `pam_test` for now, but indeed I'll use it in other packages too in an upcoming branch, so instead of making it private to rename it again in future I preferred to keep it public.

But let me know if you prefer it to be private for now and expose it later.

Reply:
Let’s make it private and just rename it in context, that’s easier when we need to dig into why.